starport-core/src/main/scala/com/krux/starport/util/ErrorHandler.scala

```diff
@@ -59,11 +54,10 @@ object ErrorHandler extends Logging with WaitForIt {
 
         // Set schedule failure count
         db.run(setScheduleFailureCountQuery).map { _ =>
-          SendEmail(
-            toEmails,
-            fromEmail,
+          Notification(pipeline.owner).send(
```

Comment:
The design here seems to be a bit redundant. `Notification(pipeline.owner)` already knows the email/sns to send, and calling `send` again requires `pipeline` for the owner again. Which might look confusing. I think we can code it in such a way that `pipeline.owner` are only needed once. Something like `Notification.send(..., ..., pipeline.owner)` or simply `Notify(..., ..., pipeline.owner)`

Reply:
I tried to make it in 2 different phases explicitly. 

Step 1: To get the notification medium. The medium can be defined by any string, it does not have to come from a field within `pipeline`.
Step 2: Actual sending. 

If we say, these steps should always be combined, then I agree.

What are your thoughts @realstraw ?

Reply:
The main drawback with the separation is `Notification(pipeline.owner)` and then calling the `send` allows inconsistent behavior that `Notification(pipeline.owner)` might be an email notification, but when calling `send`, you can in theory pass an SNS owner in the parameter.